app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -54,6 +55,8 @@ class DeadlineDetailsActivity : AppCompatActivity() {
     private var dateTime: LocalDateTime = LocalDateTime.of(2022, 10, 10, 10, 10)
     private var state: DeadlineState = DeadlineState.TODO
 
+    private lateinit var deadline: Deadline
```

Comment:
Why these changes ? It change nothing to the activity

Reply:
yeah but during unsuccessful try to add a pdf view, I had to have access to the deadline. And. I think it makes sense to have the deadline stored somewhere in a variable since the whole activity is about this deadline.

Reply:
You only need the pdf, you could have add a variable pdf to recuperate the information.  The deadline is recuperated only in `setDeadlineObserver()` as we recuperate the necessary informations in other variables. Each variables can be changed at a different time so It is not a good idea to keep all of it in a Deadline (because each time you change something, you would have to make a copy of the Deadline with the new variables wich is not efficient)
And, as we only need the full deadline here, we dont need to store it)
Also, as you tests weren't successful, it is always better to remove all unnecessary modifications